sycl/test/built-ins/printf.cpp

```diff
@@ -0,0 +1,127 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+#include <CL/sycl.hpp>
+
+#include <cstdint>
+#include <iostream>
+
+using namespace cl::sycl;
+
+// According to OpenCL C spec, the format string must be in constant address
+// space
+#ifdef __SYCL_DEVICE_ONLY__
+#define CONSTANT __attribute__((opencl_constant))
+#else
+#define CONSTANT
+#endif
+
+// This is one of the possible ways to define a format string in a correct
+// address space
+static const CONSTANT char format_hello_world[] = "Hello, World!\n";
+
+// Static isn't really needed if you define it in global scope
+const CONSTANT char format_int[] = "%d\n";
+
+static const CONSTANT char format_vec[] = "%d,%d,%d,%d\n";
+
+const CONSTANT char format_hello_world_2[] = "%lu: Hello, World!\n";
+
+int main() {
+  {
+    default_selector Selector;
+    queue Queue(Selector);
+
+    Queue.submit([&](handler &CGH) {
+      CGH.single_task<class integral>([=]() {
+        // String
+        intel::experimental::printf(format_hello_world);
+        // Due to a bug in Intel CPU Runtime for OpenCL on Windows, information
+        // printed using such format strings (without %-specifiers) might
+        // appear in different order if output is redirected to a file or
+        // another app
+        // FIXME: strictly check output order once the bug is fixed
+        // CHECK: {{(Hello, World!)?}}
+
+        // Integral types
+        intel::experimental::printf(format_int, 123);
+        intel::experimental::printf(format_int, -123);
+        // CHECK: 123
+        // CHECK-NEXT: -123
+
+        // Floating point types
+        {
+          // You can declare format string in non-global scope, but in this case
+          // static keyword is required
+          static const CONSTANT char format[] = "%f\n";
+          intel::experimental::printf(format, 33.4f);
+          intel::experimental::printf(format, -33.4f);
+        }
+        // CHECK-NEXT: 33.4
+        // CHECK-NEXT: -33.4
+
+        // Vectors
+        cl::sycl::vec<int, 4> v4{5, 6, 7, 8};
+#ifdef __SYCL_DEVICE_ONLY__
+        // On device side, vectors can be printed via native OpenCL types:
+        using ocl_int4 = cl::sycl::vec<int, 4>::vector_t;
```

Comment:
This is not intuitive, can you add this to the disclaimer in
builtins.hpp?

Reply:
This is kind of additional feature to, why do I need to put a disclaimer here?

Reply:
I suggest to document this additional feature, so people know how to use it correctly.

Reply:
Done